fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeManager.java

```diff
@@ -380,18 +500,28 @@ public static PrivilegeManager load(
                             (UserPrivilegeCollection) reader.readJson(UserPrivilegeCollection.class);
                     ret.userToPrivilegeCollection.put(userIdentity, collection);
                 }
+                // 1 json for num roles
+                int numRole = (int) reader.readJson(int.class);
+                LOG.info("loading {} roles", numRole);
+                for (int i = 0; i != numRole; ++i) {
+                    // 2 json for each role(kv)
+                    Long roleId = (Long) reader.readJson(Long.class);
+                    RolePrivilegeCollection collection =
+                            (RolePrivilegeCollection) reader.readJson(RolePrivilegeCollection.class);
+                    ret.roleIdToPrivilegeCollection.put(roleId, collection);
+                }
```

Comment:
You should update the nextRoleId here.

Reply:
nextRoleId used to store in the first part, but I have removed it anyway.